### PR TITLE
add more functionality

### PR DESCRIPTION
**Pull request by aminnj**
Starting to chip away at https://github.com/Moelf/FHist.jl/issues/13

- [x] `binedges(h::Hist1D) = h.hist.edges[1]`
- [x] `nbins(h::Hist1D)`
- [x] `bincounts(h::Hist1D) = h.hist.weights`
- [x] `integral(h::Hist1D)`
- [x] `Statistics.mean(h::Hist1D)`
- [x] `Statistics.std(h::Hist1D)`
- [x] `Statistics.quantile(h::Hist1D, p)`
- [x] `Statistics.median(h::Hist1D)`
- [x] `lookup(h::Hist1D, value)`
- [x] `cumulative(h::Hist1D; forward=true)::Hist1D`
- [x] `normalize(h::Hist1D)`

There are test cases for all of the above, and they're all quickly showcased in a new section of the readme.

Not implemented here:

* `h.rebin(3)` to combine 3 adjacent bins into one, and `h.restrict(self, low=None, high=None)` to subset the hist along the x-axis
Leave edge manipulation for another PR with dedicated discussion on indexing a histogram?

* `h.to_json()`, `h.from_json()`
Leave serialization for another PR

### CODE REVIEW

File: src/hist1d.jl [hunk 1]
```diff
@@ -11,17 +11,13 @@ Base.lock(h::Hist1D) = lock(h.hlock)
...
 
 """
-    sample(h::Hist1D)
-    sample(h::Hist1D, n::Int)
+    sample(h::Hist1D, n::Int=1)
 
-Sample a histogram's with weights equal to bin count, one or `n` times.
-The returned sample value will be one of the bin's left edge.
+Sample a histogram's with weights equal to bin count, `n` times.
+The sampled values are the bin centers.
```
Comment (Moelf) [inline src/hist1d.jl:17]: this is not how ROOT does it, IIRC ROOT always give you bin's lower edge

Comment (aminnj) [reply src/hist1d.jl:17]: Ah ok let me modify it then

File: src/hist1d.jl [hunk 2]
```diff
@@ -193,14 +221,67 @@ function Hist1D(
...
+end
+
+"""
+    normalize(h::Hist1D)
+
+Create a normalized histogram via division by `integral(h)`.
+"""
+function normalize(h::Hist1D)
+    h = deepcopy(h)
+    return h*(1/integral(h))
```
Comment (Moelf) [inline src/hist1d.jl:259]: doesn't this already return a copy?

**Comment by aminnj** [comment]
oops, thanks :)

**Review by Moelf — APPROVED** [review]

**Pull request merged**